src/components/views/dialogs/HostSignupDialog.tsx

```diff
@@ -0,0 +1,266 @@
+    }
+
+    private async sendAccountDetails() {
+        const openIdToken = await MatrixClientPeg.get().getOpenIdToken();
+        if (!openIdToken || !openIdToken.access_token) {
+            this.setState({
+                completed: true,
+                error: _t("Failed to connect to your homeserver. Please close this dialog and try again."),
+            });
+            return;
+        }
+        this.sendMessage({
+            action: PostmessageAction.HostSignupAccountDetails,
+            account: {
+                accessToken: await MatrixClientPeg.get().getAccessToken(),
+                name: OwnProfileStore.instance.displayName,
+                openIdToken: openIdToken.access_token,
+                serverName: await MatrixClientPeg.get().getDomain(),
+                userLocalpart: await MatrixClientPeg.get().getUserIdLocalpart(),
+                termsAccepted: true,
+            },
+        });
+    }
+
+    private onAccountDetailsDialogFinished = async (result) => {
+        if (result) {
+            return this.sendAccountDetails();
+        }
+        return this.closeDialog();
+    }
+
+    private onAccountDetailsRequest = () => {
+        const termsDialog = this.config.termsDialog;
+        const textComponent = (
+            <>
+                <p>
+                    {termsDialog.text}
+                </p>
+                <p>
+                    {_t("Learn more in our")}&nbsp;
+                    <a href={termsDialog.privacyPolicy.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.privacyPolicy.text}
+                    </a>,&nbsp;
+                    <a href={termsDialog.termsOfService.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.termsOfService.text}
+                    </a> {_t("and")}&nbsp;
+                    <a href={termsDialog.cookiePolicy.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.cookiePolicy.text}
+                    </a>.
+                </p>
+            </>
+        );
+        Modal.createDialog(
+            QuestionDialog,
+            {
+                title: termsDialog.title,
+                description: textComponent,
+                button: termsDialog.acceptText,
+                onFinished: this.onAccountDetailsDialogFinished,
+            },
+        );
+    }
+
+    public componentDidMount() {
+        window.addEventListener("message", this.messageHandler);
+    }
+
+    public componentWillUnmount() {
+        if (HostSignupStore.instance.isHostSignupActive) {
+            // Run the close dialog actions if we're still active, otherwise good to go
+            return this.closeDialog();
+        }
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <div className="mx_HostSignup_persisted">
+                <PersistedElement key="host_signup" persistKey="host_signup">
```

Comment:
can `host_signup` be put into a const somewhere and reused to make it feel less magical-string